### PR TITLE
messages to list communications for all Careplans, including the latest

### PR DESCRIPTION
**Pull request by achen2401**
https://www.pivotaltracker.com/story/show/184990960
With these changes, messages displayed in the Messages view will be from all care plans, not just the latest one

### CODE REVIEW

File: src/FhirClientProvider.tsx [hunk 1]
```diff
@@ -40,64 +41,90 @@ export default function FhirClientProvider(props: Props): JSX.Element {
...
-                            console.log(`Loaded ${carePlanResult.reference}`);
-                        }
-                        setLoaded(true);
-                    }, (reason: any) => setError(reason)).catch(e => {
-                        console.log("Error fetching CarePlan", e)
+                    getCarePlans(client, patientResult.id)
+                      .then(
+                        (carePlanResults: CarePlan[]) => {
+                          const activeCarePlans = carePlanResults?.filter(
+                            (item) => item.status === "active"
```
Comment (mcjustin) [inline src/FhirClientProvider.tsx:98]: @achen2401  I *think* we want all CarePlans, regardless of their `status`. But, I'm not certain, as I don't know how we manage `status` for CP's.

File: src/FhirClientProvider.tsx [hunk 2]
```diff
@@ -40,64 +41,90 @@ export default function FhirClientProvider(props: Props): JSX.Element {
...
+      }).toString();
+      return client.request(`/CarePlan?${searchParams}`).then(
+        (bundle: Bundle) => {
+          if (bundle.type === "searchset") {
+            if (!bundle.entry || !bundle.entry.length) {
+              return null;
+            }
+            if (bundle.total > 1) {
+              console.log(
+                "Multiple ISACC CarePlans found. Using the most recently updated.",
```
Comment (mcjustin) [inline src/FhirClientProvider.tsx]: This log message needs updating

File: src/FhirClientProvider.tsx (hunk 1 above)
Comment (achen2401) [reply src/FhirClientProvider.tsx:98]: thanks @mcjustin noting that, yes, the messages for all CarePlans will be displayed regardless of status, see [here](https://github.com/uwcirg/isacc-messaging-client-sof/pull/34/files#diff-ecc3420995ac53a698706c9b76aea2e203c950f91b7526d190907e893e27ae0dR132) 
The change is to get the current care plan (status == active) and also [all the careplans](https://github.com/uwcirg/isacc-messaging-client-sof/pull/34/files#diff-f985d4d496ba46e6fadc4d6a1531877caeaa241f9eccf6a885e4e6839ce8ea98R109) for the patient.

**Review by mcjustin — APPROVED** [review]

File: src/FhirClientContext.ts [hunk 3]
```diff
@@ -7,12 +7,14 @@ export type FhirClientContextType = {
     client: Client;
     patient: Patient;
     carePlan: CarePlan;
```
Comment (hannahburkhardt) [inline src/FhirClientContext.ts:9]: Could you rename this to currentCarePlan so it’s more obvious what the difference is between careplan and allcareplans?

Comment (achen2401) [reply src/FhirClientContext.ts:9]: thanks @hannahburkhardt  I will rename it.   Sorry,  I didn’t have time as I believe this is used in quite a few places.

File: src/FhirClientProvider.tsx [hunk 4]
```diff
@@ -40,64 +41,90 @@ export default function FhirClientProvider(props: Props): JSX.Element {
...
-                    setError("Unexpected resource type returned");
-                    return null;
-                }
-            } else {
-                setError("Unexpected bundle type returned");
-                return null;
+    async function getCarePlans(
+      client: Client,
+      patientId: string,
+      params: object = null
```
Comment (hannahburkhardt) [inline src/FhirClientProvider.tsx:47]: what is this for? doesn't look like the new parameter is ever used? @achen2401

Comment (achen2401) [reply src/FhirClientProvider.tsx:47]: @hannahburkhardt thank you for catching it, I will remove.  I think it is remnant during my process of refactoring the code.

**Pull request merged**